misc-scripts/xref_mapping/XrefParser/UniProtParser/Loader.pm

```diff
@@ -0,0 +1,114 @@
+=head1 LICENSE
+
+See the NOTICE file distributed with this work for additional
+information regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+
+
+package XrefParser::UniProtParser::Loader;
+
+use strict;
+use warnings;
+
+use Carp;
+
+
+sub new {
+  my ( $proto, $arg_ref ) = @_;
+
+  my $self = {
+              'batch_size'  => $arg_ref->{'batch_size'} // 1,
+              'dbh'         => $arg_ref->{'dbh'},
+              '_baseParser' => $arg_ref->{'baseParser'},
+            };
+  my $class = ref $proto || $proto;
+  bless $self, $class;
+  $self->_clear_send_buffer();
+
+  return $self;
+}
+
+
+sub baseParserInstance {
+  my ( $self ) = @_;
+  return $self->{'_baseParser'};
+}
+
+
+sub load {
+  my ( $self, $transformed_data ) = @_;
+
+  if ( ! defined $transformed_data ) {
+    return;
+  }
+
+  $self->_add_to_send_buffer( $transformed_data );
```

Comment:
I think this buffering should ultimately live in the DB abstraction class as yet to be realised, so that all parsers may buffer up their loads. I'm also not totally convinced you won't have an unflushed buffer at the end with n > 1 batch_size.

Reply:
Not by itself, no. UniProtParser::run() takes care of that by calling $loader->flush() after the loop.

Reply:
Added a call to flush() to the destructor, just in case.